### PR TITLE
Fix udp create immediate polling wrong

### PR DESCRIPTION
**Pull request by mokeyish**
This PR fixes that `self.closure` is not immediately ready.

### CODE REVIEW

File: crates/proto/src/udp/udp_stream.rs [hunk 1]
```diff
@@ -217,8 +217,11 @@ pub(crate) struct NextRandomUdpSocket<S> {
     bind_address: SocketAddr,
     closure: UdpCreator<S>,
     marker: PhantomData<S>,
+    state: Option<(usize, UdpSocketCreateFuture<S>)>,
```
Comment (bluejekyll) [inline crates/proto/src/udp/udp_stream.rs:220]: need some documentation here that the `usize` is the number of attempted socket creation attempts.

Comment (bluejekyll) [reply crates/proto/src/udp/udp_stream.rs:220]: alternatively, this could be a named type that is just an alias to `usize` to make it self-documenting.

**Review by bluejekyll — CHANGES_REQUESTED** [review]
minor change to add some documentation, otherwise, this looks like a really good change. Thank you.

**Comment by djc** [comment]
See previous PR:

- https://github.com/hickory-dns/hickory-dns/pull/2157

**Comment by djc** [comment]
(See alternative in #2464.)

**Comment by bluejekyll** [comment]
@djc, this patch looks more correct to me. then that previous patch. @mokeyish, I'm assuming this is patching an issue that's been seen in the wild, can you describe that scenario?

**Comment by djc** [comment]
Closing, since we merged #2464.

**Pull request closed without merging**